### ISSUE TITLE
Problem with eggnogmapper

### ISSUE THREAD

**Issue by snehakhokhali** [eggnogdb/eggnog-mapper#459]
Hello, I have performed functional annotation and wanted to get all the possible GO terms from the given genome of Chironomus riparius. I have only used protein sequences of CDS of the genome and run the eggnogmapper. 

Firstly, I have used protein sequences of CDS of a sample genome of D.melanogaster associated with cell cycle and run the eggnogmapper with the command "emapper.py -i flybase_DM_cyc_protein.faa -o test_cyc --override -m diamond --sensmode fast --go_evidence non-electronic". I have got the file as output "test_cyc.emapper.annotations". And I have counted all the GO IDs that I have got from the output file. Actually I wanted to count the number of GO ID (GO:0044848) i.e. biological phase and particularly GO:0022403 i.e. cell cycle phase. But I got 0 counts. 
I have also tried with --sensmode sensitive but it gave me same results.

And I have tried same with the genome of Chironomus riparius genome and it also gives 0 counts. 

Why am I not getting the counts of those IDs of child nodes of biological process??

**Comment by Cantalapiedra** [comment]
Hi @snehakhokhali ,

Did you try with `--go_evidence all`?

**Comment by Cantalapiedra** [comment]
No. Actually, I did a search and GO:0044848 is present in the database but GO:0022403 is not.

**Comment by snehakhokhali** [comment]
Could you please make an effort to include the GO term  (GO:0022403) somehow in the database? I need the cell cycle phase for my project.

**Comment by Cantalapiedra** [comment]
Hi @snehakhokhali ,

If I could do it, I would be glad to help. But unfortunately eggNOG DB is complex and a lot of effort is done to release each new version.

Sorry for the inconveniences.

Did you try annotating your proteins using some species (or related-species) reference or a similar strategy? 

Best,
Carlos

**Comment by snehakhokhali** [comment]
Sorry, I didn't get your question.

**Comment by Cantalapiedra** [comment]
Sorry, I will try to say in other words. 

If understand correctly, you are trying to annotate those GO terms in your proteins, being the latter from specific eukaryotic genomes. If that is the case, you may try mapping your proteins to a reference, already annotated, genome, from a related species. For instance, you may map your proteins against a Drosophila reference genome and retrieve the annotations, including GO terms, from those mappings... But this is just a suggestion that may or may not fit your project goals.

Best,
Carlos

**Comment by Cantalapiedra** [comment]
Hi,

Sorry, I don't know much about the available resources for the genome you are working with.

If you have a reference genome, which is already annotated with GOs, you may retrieve those GOs from the reference after mapping your proteins to it. 

If the reference genome doesn't have annotated GOs, then it is a different picture, and you may need to rely on other strategies, like looking into other similar genomes which are annotated with GOs, or trying to annotate your proteins with eggNOG (which we already know it is not working for your goals), or mapping your proteins to Uniprot, nr, or other means and DBs which I am not aware of.

**Comment by snehakhokhali** [comment]
I am performing the functional annotation. I have got the fasta file of genome of Chironomus riparius and gff file retrieved from MAKER (structural annotation). I am trying to annotate the protein with eggNOG so that I can retrieve GO terms. And I need the GO terms of biological process. But unfortunately there are no GO terms of cell cycle phase in eggNOG. Can you suggest some paper which doesn't have annotated GOs in reference genome?

**Comment by Cantalapiedra** [comment]
Is there any other closely related-species which has an already annotated reference genome? (I don't know if D. melanogaster or other species are close enough to Chironomus riparius or not).
You may check Ensembl genomes or other sources of reference genomes, which usually include GO annotations.

Another way to try, maybe mapping your proteins to UniprotKB and try to annotate using the GO terms associated to Uniprot entries. Note that when you check the gene products of GO terms (http://amigo.geneontology.org/amigo/search/bioentity?q=GO%3A0022403&searchtype=geneproduct) the entries are proteins from DB like Uniprot, FlyBase, etc. For isntance, there are 9 proteins annotated as GO:0022403 for D. melanogaster in the previous link.

I don't know about current papers on these topics, but, yes, as you said and you sure already did, I would try looking for recent papers describing genome annotation of eukaryotic species similar to yours, to get ideas.

**Comment by snehakhokhali** [comment]
thank you for the suggestion.

**Comment by Cantalapiedra** [comment]
good luck!
